docker/docker-compose.yml

```diff
@@ -0,0 +1,98 @@
+ambassador:
+  image: cpuguy83/docker-grand-ambassador
+  volumes:
+  - /var/run/docker.sock:/var/run/docker.sock
+  command: "-name docker_etcd_1 -name docker_mesos_1 -name docker_apiserver_1 -name docker_controller_1 -name docker_scheduler_1"
+etcd:
+  hostname: etcd
+  image: coreos/etcd
+  ports: [ "4001:4001", "7001:7001" ]
+  command: -addr=etcd:4001
+#zookeeper:
+#  image: jplock/zookeeper:3.4.5
+#  ports: [ "2181:2181" ]
+mesos:
+  hostname: mesos
+  image: mesosphere/mesos:0.22.0-1.0.ubuntu1404
+  entrypoint: mesos-local
+  ports: [ "5050:5050" ]
+  environment:
+  - MESOS_HOSTNAME=mesos
+  - MESOS_PORT=5050
+  - MESOS_LOG_DIR=/var/log/mesos
+  - MESOS_QUORUM=1
+  - MESOS_REGISTRY=in_memory
+  - MESOS_WORK_DIR=/var/lib/mesos
+#  - MESOS_ZK=zk://zookeeper:2181/mesos
+  links:
+  - etcd
+  - "ambassador:apiserver"
+  volumes:
+  - /var/run/docker.sock:/var/run/docker.sock
+#  - zookeeper
+apiserver:
+  hostname: apiserver
+  image: mesosphere/kubernetes-mesos:latest
+  entrypoint:
+  - /bin/bash
+  - "-c"
+  - >
+    echo "Hostname: $(hostname -f) ($(hostname -f | xargs resolveip))" &&
+    echo "mesos-master = mesos:5050" >> /opt/mesos-cloud.conf &&
+    await-service $(resolveip etcd) 4001 &&
+    await-service $(resolveip mesos) 5050 &&
+    km apiserver
+    --address=$(resolveip apiserver)
+    --external_hostname=apiserver
+    --etcd_servers=http://etcd:4001
+    --portal_net=10.10.10.0/24
+    --port=8888
+    --cloud_provider=mesos
+    --cloud_config=/opt/mesos-cloud.conf
+    --v=2
+  ports: [ "8888:8888" ]
+  links:
+  - etcd
+  - mesos
+controller:
+  hostname: controller
+  image: mesosphere/kubernetes-mesos:latest
+  entrypoint:
+  - /bin/bash
+  - "-c"
+  - >
+    echo "Hostname: $(hostname -f) ($(hostname -f | xargs resolveip))" &&
+    echo "mesos-master = mesos:5050" >> /opt/mesos-cloud.conf &&
+    await-service $(resolveip mesos) 5050 20 &&
+    await-service $(resolveip apiserver) 8888 &&
+    km controller-manager
+    --master=http://apiserver:8888
+    --cloud_config=/opt/mesos-cloud.conf
+    --v=2
+  links:
+  - mesos
+  - apiserver
+scheduler:
+  hostname: scheduler
+  image: mesosphere/kubernetes-mesos:latest
+  entrypoint:
+  - /bin/bash
```

Comment:
The entrypoint in string format is always run inside a shell by Docker. I guess there is no need for another bash line here. Completely sure yet, especially about the way docker-compose handles this.